boot/bootutil/src/loader.c

```diff
@@ -416,8 +417,14 @@ boot_read_status_bytes(const struct flash_area *fap, struct boot_status *bs)
 
     if (found) {
         i--;
-        bs->idx = i / BOOT_STATUS_STATE_COUNT;
-        bs->state = i % BOOT_STATUS_STATE_COUNT;
+        /* FIXME: is this test required? */
```

Comment:
That the resulting values of the division/modulus give the same values that are hardcoded in the `else` so it ends up not being required. It was just a reminder to check later and clean up if that's actually possible.